vendor/github.com/Microsoft/hcsshim/interface.go

```diff
@@ -17,6 +17,11 @@ type MappedPipe = schema1.MappedPipe
 type HvRuntime = schema1.HvRuntime
 type MappedVirtualDisk = schema1.MappedVirtualDisk
 
+// AssignedDevice represents a device that has been directly assigned to a container
+//
+// NOTE: Support added in RS5
+type AssignedDevice = schema1.AssignedDevice
```

Comment:
I just noticed the `-1` in this release, and wondered why; that's when I noticed this was added in v0.7.9-1 https://github.com/Microsoft/hcsshim/blob/v0.7.9-1/interface.go#L20-L23, but;

- not present in v0.7.10 https://github.com/Microsoft/hcsshim/blob/v0.7.10/interface.go
- not present in v0.7.11 https://github.com/Microsoft/hcsshim/blob/v0.7.11/interface.go
- not present in v0.7.12 https://github.com/Microsoft/hcsshim/blob/v0.7.12/interface.go
- added again in v0.7.12-1 https://github.com/Microsoft/hcsshim/blob/v0.7.12-1/interface.go#L20-L23
- not present in v0.7.13 https://github.com/Microsoft/hcsshim/blob/v0.7.13/interface.go
- added again in v0.7.14 https://github.com/Microsoft/hcsshim/blob/v0.7.14/interface.go#L20-L23
- present in all 0.8.x versions

Curious; was it not possible to either update to (e.g.) a new `v0.7.x` version (`v0.7.15`), or update to `v0.8.x`? (I assumed the hcsshim repository used SemVer, so updating to newer patch releases wouldn't break things)

Reply:
Unfortunately hcsshim was not using SymVer correctly at the time of these releases. The minimal change to docker was to make this spot fix to the v0.7.9 release + this one commit. This change is already part of master in the v0.8.x releases as you see and the v0.8.x releases are properly using SymVer from now on. When we update docker to one of those releases (much bigger change) we will no longer need this spot fix.

Reply:
Well, strictly it's still 0.x, so breaking changes are allowed 😅 

My concern was that if there's bugfixes in hcsshim, and more recent (0.7.x) versions have breaking changes, we wouldn't be able to update to those versions, unless those fixes are backported to (e.g.) `v0.7.12-2`.

Are there any breaking changes between `v0.7.12` and `v0.7.14` that currently prevent us from updating to the latest 0.7.x version? https://github.com/Microsoft/hcsshim/compare/v0.7.9...v0.7.14

If there are; are those changes addressable in the moby codebase?

Reply:
@jhowardmsft - Is likely better equip to answer that question. My assumption is that if we need a v0.7 release we will have to carry this fix as you say. But hopefully when @jhowardmsft  is ready we will just move to the v0.8.x releases. I am not aware of any breaking changes for the v1 API's that Docker uses